### PR TITLE
Bugfix/listsegment test memory

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR addresses two issues, one involving memory usage in indexset tests, and one involving memory management in the list segment copy ctor.
- The indexset tests now pass with all back-ends, in particular with openmp target.

### CODE REVIEW

**Comment by rhornung67** [comment]
@trws I think we agreed at a meeting a while back to deprecate the compile-time selection of UM usage in ListSegment (depends on whether CUDA or HIP is enabled) at the next release, then remove it after that (tests, examples, perf suite, proxy apps, etc. will all need to be adjusted) I know there is clean up to do in the ListSegment implementation. I did something fairly quick and dirty to start using camp resources in the tests.

File: test/include/RAJA_test-indexset-build.hpp [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 #include "camp/resource.hpp"
 
+#include <random>
```
Comment (rchen20) [inline test/include/RAJA_test-indexset-build.hpp:19]: Extra include here? Otherwise, this looks good.

**Comment by trws** [comment]
We did, but given that camp resources are currently allocating managed anyway, we could cull out a decent chunk of the compile time selection stuff and replace it with a default camp resource based on whether we find cuda or hip support, then later require the resource as an argument.  It should reduce a bunch of the code out but give us the same behavior until we're ready to actually break the API.

File: test/include/RAJA_test-indexset-build.hpp (hunk 1 above)
Comment (rhornung67) [reply test/include/RAJA_test-indexset-build.hpp:19]: At one point, I was trying to make the header files self-contained so they're easier to use; i.e., if you use them, you don't have to worry about include order, etc.. But then I didn't follow through all the way.  I'll do a clean up pass on this in a separate PR.

**Pull request merged**